src/effects.js

```diff
@@ -153,14 +154,11 @@ function defaultPrefilter( elem, props, opts ) {
 
 		// Identify a display type, preferring old show/hide data over the CSS cascade
 		restoreDisplay = dataShow && dataShow.display;
-		if ( restoreDisplay == null ) {
-			restoreDisplay = dataPriv.get( elem, "display" );
-		}
+
 		display = jQuery.css( elem, "display" );
```

Comment:
There seems to be some confusion about the perf issue. Retrieving display from computed CSS is the problem. Webkit takes a long time retrieving that value. Our defaultDisplay implementation was not the problem.